serverlessworkflow/modules/ROOT/pages/eventing/working-with-openapi-callbacks.adoc

```diff
@@ -0,0 +1,121 @@
+
+For the workflow to identify the published CloudEvent it is waiting for, the external service developer includes the workflow instance ID in the CloudEvent header or uses the xref:eventing/event-correlation-with-workflows.adoc[Event correlation]. The following figure displays the process:
+
+image::eventing/callbacks-explained.png[]
+
+The workflow correlation described in this document focuses on the former mechanism that is based on the fact that each workflow instance contains a unique identifier, which is generated automatically.
+
+[[ref-example-callback]]
+== Example of the OpenAPI Callback
+
+To understand the OpenAPI Callback, see the link:{kogito_sw_examples_url}/serverless-workflow-callback-events-over-http-quarkus[`serverless-workflow-callback-events-over-http-quarkus`] example application in GitHub repository.
+
+The `serverless-workflow-callback-events-over-http-quarkus` application is initiated with the following request:
+[code,json]
+----
+{
+  "message": "Hello"
+}
+----
+Once the workflow is started, it makes an external service call with the callback URL and the workflow instance ID in the request body. Then, as configured in the link:{kogito_sw_examples_url}/serverless-workflow-callback-events-over-http-quarkus/callback-workflow/src/main/resources/specs/callback.yaml[OpenAPI] file, the callback URL is invoked to send a CloudEvent to the workflow.
+
+Once the `wait` type CloudEvent is received, the workflow moves to the next state and ends successfully. The following figure shows the link:{kogito_sw_examples_url}/serverless-workflow-callback-events-over-http-quarkus[`serverless-workflow-callback-events-over-http-quarkus`] image:
+
+image::eventing/openapi-callback.png[]
+
+To use the OpenAPI callback in a workflow, the OpenAPI YAML file is configured with the callback as specified in the link:{open_api_spec_url}#callback-object-examples[Openapi callback object example].
+
+To use the Callback state in a workflow, first CloudEvent type `wait` is declared that the workflow uses. Following is an example of CloudEvents declaration in a workflow definition:
+
+.Example of CloudEvent declaration in a workflow definition
+[code,json]
+----
+ "events": [
+    {
+      "name": "waitEvent",
+      "source": "",
+      "type": "wait"
+    }
+----
+
+After that, a Callback state is declared, and it waits for a CloudEvent with the `wait` type. Following is an example of declaring a Callback state that handles the `wait` type CloudEvent:
+
+.Example of a Callback State declaration handling the `wait` CloudEvent
+[code,json]
+----
+ {
+      "name": "waitForEvent",
+      "type": "callback",
+      "action":
+        {
+        "functionRef": {
+          "refName": "callBack",
+          "arguments": {
+            "uri": "http://localhost:8080/wait",
```

Comment:
I think we should explain as a TIP or NOTE how to set this URL dynamically via an env var ?